From 51768c1446808f17e3ae7ded1dcd75f6e5f3d561 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Wed, 18 Oct 2017 17:08:13 +0200
Subject: [PATCH] Avoid bumping the GLib version

Commit ba9cc7bf2af77b4 add a Todoist provider that is not used in RHEL.
Therefore, one can avoid requiring a GLib 2.52 by not using
g_uuid_string_random.

Don't use this patch if Todoist is ever enabled.

https://bugzilla.redhat.com/show_bug.cgi?id=1503726
---
 configure.ac                        | 2 +-
 src/goabackend/goatodoistprovider.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 520ad03b0a8d..10126db02cee 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,7 +85,7 @@ GTK_DOC_CHECK([1.3])
 # Libraries
 #
 
-PKG_CHECK_MODULES(GLIB, [glib-2.0 gio-2.0 gio-unix-2.0 >= 2.52])
+PKG_CHECK_MODULES(GLIB, [glib-2.0 gio-2.0 gio-unix-2.0 >= 2.44])
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
diff --git a/src/goabackend/goatodoistprovider.c b/src/goabackend/goatodoistprovider.c
index f0d3e43602d8..bcfa7cd94882 100644
--- a/src/goabackend/goatodoistprovider.c
+++ b/src/goabackend/goatodoistprovider.c
@@ -116,7 +116,7 @@ build_authorization_uri (GoaOAuth2Provider  *oauth2_provider,
   gchar *state;
   gchar *uri;
 
-  state = g_uuid_string_random ();
+  state = g_strdup ("this-is-not-a-real-uuid");
   uri = g_strdup_printf ("%s"
                          "?response_type=token"
                          "&client_id=%s"
-- 
2.14.3

